packages/core/src/mesh/ModelMesh.ts

```diff
@@ -14,4 +66,793 @@ export class ModelMesh extends Mesh {
+   */
+  uploadMeshData(noLongerAccessible: boolean): void {
+    const { _indicesArray, _vertexSlotChanged, _accessible } = this;
+    if (!_accessible) {
+      throw "Not allowed to access data while accessible is false.";
+    }
+
+    // Structure of vertex buffer has changed
+    if (_vertexSlotChanged || this._vertexCountChanged) {
+      const vertexElements = this._updateVertexElements();
+      const elementCount = this._elementCount;
+      const vertices = new Float32Array(elementCount * this.vertexCount);
+      this._verticesArray = vertices;
+      this._vertexChangeFlag = ValueChanged.All;
+      this._updateVertices(vertices);
+      this._vertexBuffer = new Buffer(
+        this._engine,
+        BufferBindFlag.VertexBuffer,
+        vertices,
+        noLongerAccessible ? BufferUsage.Static : BufferUsage.Dynamic
+      );
+      this.setVertexElements(vertexElements);
+      this.setVertexBufferBinding(new VertexBufferBinding(this._vertexBuffer, elementCount * 4));
+      this._vertexSlotChanged = false;
+      this._vertexCountChanged = false;
+    } else {
+      const verticesArray = this._verticesArray;
+      this._updateVertices(verticesArray);
+      this._vertexBuffer!.setData(verticesArray);
+    }
+
+    if (_indicesArray) {
+      if (!this._indexBuffer || _indicesArray.byteLength != this._indexBuffer.byteLength) {
+        this._indexBuffer = new Buffer(this._engine, BufferBindFlag.IndexBuffer, _indicesArray);
+        this.setIndexBufferBinding(this._indexBuffer, this._indicesFormat);
+      } else if (this._indicesChangeFlag) {
+        this._indicesChangeFlag = false;
+        this._indexBuffer.setData(this._indicesArray);
+        if (this._indexBufferBinding._format !== this._indicesFormat) {
+          this.setIndexBufferBinding(this._indexBuffer, this._indicesFormat);
+        }
+      }
+    } else if (this._indexBuffer) {
+      this.setIndexBufferBinding(null);
+      this._indexBuffer = null;
+    }
+
+    if (noLongerAccessible) {
+      this._releaseCache();
+    }
+  }
+
+  /**
+   * Set indices for the mesh.
+   * @param indices - The indices for the mesh.
+   */
+  setIndices(indices: Uint8Array | Uint16Array | Uint32Array): void {
+    if (!this._accessible) {
+      throw "Not allowed to access data while accessible is false.";
+    }
+    this._indicesArray = indices;
+    if (indices === null) {
+      return;
+    }
+
+    if (indices instanceof Uint8Array) {
+      this._indicesFormat = IndexFormat.UInt8;
+    } else if (indices instanceof Uint16Array) {
+      this._indicesFormat = IndexFormat.UInt16;
+    } else if (indices instanceof Uint32Array) {
+      this._indicesFormat = IndexFormat.UInt32;
+    }
+    this._indicesChangeFlag = true;
+  }
+
+  /**
+   * Get array of indices for the mesh.
+   */
+  getIndices(): Uint8Array | Uint16Array | Uint32Array {
+    return this._indicesArray;
+  }
+
+  /**
+   * Set positions array for the mesh.
```

Comment:
It think the commnets use"Set positions for the mesh." is better. No need to write "array",array is the data type,developer can know by TypeScript type system.

Other comments are the same.